src/clients/python/experimental_api_v2/library/cuda_shared_memory/__init__.py

```diff
@@ -114,14 +116,15 @@ def get_raw_handle(cuda_shm_handle):
     
     Returns
     -------
-    bytes
-        The underlying raw serialized cudaIPC handle
+    bytes 
+        The raw serialized cudaIPC handle of underlying cuda shared memory
+        in base64 encoding
 
     """
     craw_handle = c_char_p()
     _raise_if_error(c_int(_ccudashm_get_raw_handle(cuda_shm_handle, byref(craw_handle))))
 
-    return craw_handle.value
+    return base64.b64decode(craw_handle.value.decode())
```

Comment:
Why are you encoding it on the C side just to immediately decode it here? Why not just leave it as raw bytes. The return description also seems wrong.

Reply:
Ahh, I misinterpreted your earlier suggestion. Using base64 was a workaround for ctypes. The datatypes were null-terminated when passed from c to python.

Fixed as per your original suggestion.